source/frontend/src/features/mapSideBar/acquisition/AcquisitionRouter.tsx

```diff
@@ -0,0 +1,98 @@
+import { FormikProps } from 'formik';
+import React from 'react';
+import { Redirect, Route, Switch, useRouteMatch } from 'react-router-dom';
+
+import { InventoryTabNames } from '@/features/mapSideBar/property/InventoryTabs';
+import { FileTabType } from '@/features/mapSideBar/shared/detail/FileTabs';
+import { Api_AcquisitionFile } from '@/models/api/AcquisitionFile';
+import { stripTrailingSlash } from '@/utils';
+
+import { EditFormType } from './EditFormNames';
+import { AcquisitionFileTabs } from './tabs/AcquisitionFileTabs';
+import { UpdateAgreementsContainer } from './tabs/agreement/update/UpdateAgreementsContainer';
+import { UpdateAgreementsForm } from './tabs/agreement/update/UpdateAgreementsForm';
+import { UpdateAcquisitionChecklistContainer } from './tabs/checklist/update/UpdateAcquisitionChecklistContainer';
+import { UpdateAcquisitionChecklistForm } from './tabs/checklist/update/UpdateAcquisitionChecklistForm';
+import { UpdateAcquisitionContainer } from './tabs/fileDetails/update/UpdateAcquisitionContainer';
+import { UpdateAcquisitionForm } from './tabs/fileDetails/update/UpdateAcquisitionForm';
+import { UpdateStakeHolderContainer } from './tabs/stakeholders/update/UpdateStakeHolderContainer';
+import { UpdateStakeHolderForm } from './tabs/stakeholders/update/UpdateStakeHolderForm';
+
+export interface IAcquisitionRouterProps {
+  formikRef: React.Ref<FormikProps<any>>;
+  acquisitionFile?: Api_AcquisitionFile;
+  isEditing: boolean;
+  setIsEditing: (value: boolean) => void;
+  activeEditForm?: EditFormType;
+  defaultFileTab: FileTabType;
+  defaultPropertyTab: InventoryTabNames;
+  onSuccess: () => void;
+}
+
+export const AcquisitionRouter: React.FC<IAcquisitionRouterProps> = props => {
+  const { path, url } = useRouteMatch();
+
+  if (props.acquisitionFile === undefined || props.acquisitionFile === null) {
+    return null;
+  }
+
+  // render edit forms
+  if (props.isEditing) {
```

Comment:
what should we do if we are editing and our path doesn't match one of these? currently we display an empty edit screen.

Reply:
The old code in ViewSelector throws an error - `throw Error('Active edit form not defined');`
I can create a default route that does the same

Reply:
Fixed. We redirect to the default File tab (file details)